docker/README.md

```diff
@@ -0,0 +1,105 @@
+
+While it is technically possible to run docker within a docker container, doing so requires running the mesos (slave) in priveledged mode.
+So the simplest way is to instead expose the bind mount the docker socket (`-v "/var/run/docker.sock:/var/run/docker.sock"`)
+
+In addition to running in docker, it's also possible to [run docker locally](#local).
+
+<a name="complete"/>
+## K8SM Complete
+
+The "complete" Dockerfile includes everything needed to run a development instance of kubernetes-mesos, including etcd and mesos.
+
+### Background mode
+
+```
+docker run --name kubernetes-mesos -p 8888:8888 -p 5050:5050 -p 4001:4001 mesosphere/kubernetes-mesos &> /tmp/kubernetes-mesos-docker.log &
+```
+
+To attach in interactive mode to a container already running in background mode, use:
+
+```
+docker exec -it kubernetes-mesos /bin/bash
+```
+
+### Interactive mode
+
+```
+docker run -it --name kubernetes-mesos -p 8888:8888 -p 5050:5050 -p 4001:4001 --entrypoint=/bin/bash mesosphere/kubernetes-mesos
+```
+
+Note: Interactive mode launches bash instead of the start script.
+
+### Stopping
+
+```
+docker kill kubernetes-mesos
+```
+
+### Building
+
+```
+# from root of project (not inside the docker dir)
+docker build -f ./docker/complete/Dockerfile -t mesosphere/kubernetes-mesos-complete .
+```
+
+
+<a name="composed"/>
+## K8SM Composed
+
+This method requires [Docker Compose](https://docs.docker.com/compose/) to be installed, which can be done via apt-get, homebrew, or [manually](https://docs.docker.com/compose/install/).
+
+The provided docker-compose.yml contains a self-contained configuration for running kubernetes-mesos, including its dependencies (etcd & mesos).
+It will launch 5 docker containers linked together with hostnames and port forwarding.
+
+```
+# from inside the docker dir
+docker-compose up
+```
+
+This will tail all the logs for each of the containers into STDOUT.
+`ctrl-C` to exit.
+`docker-compose rm` to remove them afterwards (required to be able to start them again).
+
+### Building
+
+The docker-compose.yml file references multiple different docker images.
+
+The one that contains just kubernetes-mesos (not mesos or etcd) can be built with the following:
+
+```
+# from root of project (not inside the docker dir)
+docker build -f ./docker/complete/Dockerfile -t mesosphere/kubernetes-mesos-complete .
+```
+
+
+<a name="local"/>
+## K8SM Local
+
+Since kubernetes-mesos requires etcd, mesos, and docker, these can all be started at once or individually.
+
+To start etcd, mesos, and kubernetes-mesos to run locally use the following command:
+
+```
+$ ./docker/bin/km-complete
+```
+
+If you already have etc and mesos running locally use the following to start just the three kubernetes-mesos components:
+
+```
+$ ./docker/bin/km-local
+```
+
+Notes:
+- The kubernetes executor, and thus the mesos slave, needs docker to be installed (or configured via tcp). By default it talks to it using the socket `/var/run/docker.sock`.
```

Comment:
More like the other way around, the mesos-slave needs docker (the containerizer actually) and therefore the k8sm executor as well which makes use of the containerizer of the slave.